test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +91,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (new Parser().parseCommand(inputCommand).isMutating()) {
+            assertEquals(addressBook, saveFile.load());
```

Comment:
To be complete, you should have an `else` branch that confirms the file was not updated for non mutating commands (e.g. by checking the last updated time of the file).